framework/include/outputs/OutputWarehouse.h

```diff
@@ -189,6 +189,43 @@ class OutputWarehouse
    */
   void bufferConsoleOutputsBeforeConstruction(bool buffer) { _buffer_action_console_outputs = buffer; }
 
+public:
+  /**
+   * Calls the initialSetup function for each of the output objects
+   * @see FEProblemBase::initialSetup()
+   */
+  void initialSetup();
```

Comment:
If making them public avoids "friend", then that's the right thing to do. I just wanted to understand why.